tools/release/changelog.ts

```diff
@@ -68,45 +76,92 @@ export async function promptChangelogReleaseName(): Promise<string> {
+    // by package names.
+    mainTemplate: readFileSync(join(__dirname, 'changelog-root-template.hbs'), 'utf8'),
+    commitPartial: readFileSync(join(__dirname, 'changelog-commit-template.hbs'), 'utf8'),
+
     // Specify a writer option that can be used to modify the content of a new changelog section.
     // See: conventional-changelog/tree/master/packages/conventional-changelog-writer
     finalizeContext: (context: any) => {
-      context.commitGroups = context.commitGroups.filter((group: any) => {
-        group.commits = group.commits.filter((commit: any) => {
-
-          // Commits that change things for "cdk-experimental" or "material-experimental" will also
-          // show up in the changelog by default. We don't want to show these in the changelog.
-          if (commit.scope && commit.scope.includes('experimental')) {
-            console.log(yellow(`  ↺   Skipping experimental: "${bold(commit.header)}"`));
-            return false;
-          }
+      const packageGroups: {[packageName: string]: any[]} = {};
 
+      context.commitGroups.forEach((group: any) => {
+        group.commits.forEach((commit: any) => {
           // Filter out duplicate commits. Note that we cannot compare the SHA because the commits
           // will have a different SHA if they are being cherry-picked into a different branch.
           if (existingChangelogContent.includes(commit.subject)) {
             console.log(yellow(`  ↺   Skipping duplicate: "${bold(commit.header)}"`));
             return false;
           }
-          return true;
+
+          // Commits which just specify a scope that refers to a package but do not follow
+          // the commit format that is parsed by the conventional-changelog-parser, can be
+          // still resolved to their package from the scope. This handles the case where
+          // a commit targets the whole package and does not specify a specific scope.
+          // e.g. "refactor(material-experimental): support strictness flags".
+          if (!commit.package && commit.scope) {
+            const matchingPackage = releasePackages.find(pkgName => pkgName === commit.scope);
+            if (matchingPackage) {
+              commit.scope = null;
+              commit.package = matchingPackage;
+            }
+          }
+
+          // TODO(devversion): once we formalize the commit message format and
+          // require specifying the "material" package explicitly, we can remove
+          // the fallback to the "material" package.
+          const packageName = commit.package || 'material';
+          const {color, title} = getTitleAndColorOfTypeLabel(commit.type);
+
+          if (!packageGroups[packageName]) {
+            packageGroups[packageName] = [];
+          }
+
+          packageGroups[packageName].push({
+            typeDescription: title,
+            typeImageUrl: `https://img.shields.io/badge/-${title}-${color}`,
```

Comment:
@jelbourn This is just a proposal. When I tried preserving the old "design"/formatting with multiple sub-sections it really looked confusing and over-complicated the whole changelog visually IMO.

e.g.

```
### cdk
#### Features
....
#### Bug fixes
....
````

Let me know what you think. I attached a picture of how it could look. If we want to go with something like that, I'm not entirely sure whether we want to have an external path inside the changelog.

Reply:
I like the flatter table structure overall, I just left some other comments about the specifics